core/headerchain.go

```diff
@@ -512,3 +512,27 @@ func (hc *HeaderChain) Engine() consensus.Engine { return hc.engine }
 func (hc *HeaderChain) GetBlock(hash common.Hash, number uint64) *types.Block {
 	return nil
 }
+
+// GetAncestorHash return the hash of ancestor at the given number
+// This implementation is slow, since it does not use a hashstorage
+// lookup like blockchain. However, the light clients does not require
+// fast lookups for EVM execution. This should probably be improved if this
+// method becomes more used.
+func (hc *HeaderChain) GetAncestorHash(ref *types.Header, target uint64) common.Hash {
+	number := ref.Number.Uint64() - 1
```

Comment:
I haven't checked if this method is ever used on `HeaderChain` (or `LightChain`, but if it's not then I would consider not including this code. The only place I could see it was maybe used was to adhere to the `ChainContext` interface, but neither of these chains support transaction processing (which is what the `ChainContext` says it is for).

Reply:
No, they're not used, but they had to have it to provide `ChainContext`. I didn't investigate further why, but these are totally unused methods

Reply:
Light clients can actually execute transactions on demand (i.e. CALL via RPC). It might be slow since it needs to pull state from remote nodes, but it should nonetheless be able to, so you do need the method.